ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -436,109 +440,10 @@ addBlock cfg blk m = Model {
---    2.  Validate first chain C0 resulting in C0V and C0I.
---    3.  Append C0V to the list of remaining candidates: [C1, C2] ++ [C0V]
---    4.  Add the valid blocks to the state:
---        > knownValid = append C0V knownValid
---    5.  Add the invalid blocks to the state:
---        > knownInvalid = append C0I knownInvalid
---    6.  Re-sort list
---        > sortBy `selectView` [C1, C2, C0V] == [C1, C0V, C2]
---    7.  Validate first chain C1 resulting in C1V and C1I.
---    8.  Append C1V to the list of remaining candidates: [C0V, C2] ++ [C1V]
---    9.  Add the valid blocks to the state:
---        > knownValid   = append C1V knownValid
---    10. Add the invalid blocks to the state:
---        > knownInvalid = append C1I knownInvalid
---    11. Re-sort list
---        > sortBy `selectView` [C0V, C2, C1V] == [C1V, C0V, C2]
---    12. Validate first chain C1V, which is fully valid and returned.
---
--- 3. If such a candidate is found, the algorithm will return it as a result.
---    Otherwise, the algorithm will return a 'Nothing'.
---
---    > chainSelection [C0, C1, C2] = Just C1V
---
--- On the other hand, the chain selection on the model takes some shortcuts to
--- achieve the same result:
---
--- 1. 'validChains' will return the list of candidates sorted by 'SelectView' and
---    each candidate is truncated to its valid prefix.
---
---    > validChains [C0, C1, C2] = (invalid == C0I + C1I, candidates == [C0V, C1V, C2])
---
--- 2. 'selectChain' will sort the chains by 'SelectView' but note that now it will
---    use the 'SelectView' of the already truncated candidate.
---
---    > selectChain [C0V, C1V, C2] = listToMaybe (sortBy `selectView` [C0V, C1V, C2])
---    >                            = listToMaybe ([C1V, C0V, C2])
---    >                            = Just C1V
---
---    The selected candidate will be the same one that the chain selection
---    algorithm would choose. However, as the chain selection algorithm will
---    consider the candidates as they were sorted by 'SelectView' on the
---    non-truncated candidates, blocks in 'C0V' are also considered valid by the
---    real algorithm.
---
---    To get as a result a set of valid blocks that mirrors the one from the
---    real algorithm, the model can process the list of candidates returned by
---    'validChains' until it find the one 'selectChain' chose as these will be
---    the ones that the real algorithm would test and re-add to the list once
---    truncated.
---
---    > knownInvalid = append (C0I + C1I) knownInvalid
---    > knownValid   = foldl append knownValid (takeWhile (/= C1V) candidates ++ [C1V])
---
---    Note that the set of known valid blocks is equivalent to the set computed
---    by real algorithm, but the set of known invalid blocks is a superset of
---    the ones known by the real algorithm. See the note
---    Ouroboros.Storage.ChainDB.StateMachine.[Invalid blocks].
+               (map fst candidates)
```

Comment:
As `valid m` is cumulative, shouldn't it always be enough to use `consideredCandidates` here? Both the model and the SUT will only ever validate blocks that are on a chain that extends the immutable chain.

I am asking this both as this is a difference to #284, and as `consideredChains` is currently only used as an input to `selectChain`, but not to build `valid'`.